src/core/tranche/operator.sol

```diff
@@ -0,0 +1,129 @@
+    function repay(address, uint) public;
+    function borrow(address, uint) public;
+}
+
+contract QuantLike {
+   uint public debt; 
+   uint public iBorrow;
+   function file(bytes32, uint) public;
+   function updateDebt(int) public;
+   function UpdateiBorrow(uint, uint) public;
+   function getSpeed() public returns(uint);
+}
+
+contract SlicerLike {
+   function calcSlice(uint) public returns(uint);
+   function calcPayout(uint) public returns(uint);
+   function updateISupply(uint, uint, uint) public;
+}
+
+// Operator
+// Manages the reserve. Triggers iSupply & iBorrow calculations.
+contract Operator is DSNote {
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address usr) public auth note { wards[usr] = 1; }
+    function deny(address usr) public auth note { wards[usr] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+
+    // --- Data ---
+    ReserveLike public reserve;
+    QuantLike public quant;
+    SlicerLike public slicer;
+
+    bool public supplyActive;
+    bool public redeemActive;
+
+    constructor( address reserve_, address quant_, address slicer_) public {
+        wards[msg.sender] = 1;
+        quant = QuantLike(quant_);
+        slicer = SlicerLike(slicer_);
+        reserve = ReserveLike(reserve_);
+        supplyActive = true;
+        redeemActive = true;
+    }
+
+    function file(bytes32 what, bool data) public note auth {
+        if (what == "supply") { supplyActive = data; }
+        else if (what == "redeem") { redeemActive = data; }
+    }
+    
+    function balance() public returns (uint) {
+        return reserve.balance();
+    }
+
+    function supply(address usr, uint currencyAmount) public note auth {
+        require (supplyActive);
+        uint tokenAmount = slicer.calcSlice(currencyAmount);
+        reserve.supply(usr, tokenAmount, currencyAmount);
+        updateSupplyRate();
+    }
+
+    function redeem(address usr, uint tokenAmount) public note auth {
+        require (redeemActive);
+        uint slice = reserve.sliceOf(usr); 
+         if (slice < tokenAmount) {
+            tokenAmount = slice;
+        }
+        uint currencyAmount = slicer.calcPayout(tokenAmount);
+        reserve.redeem(usr, tokenAmount, currencyAmount);
+        updateSupplyRate();
+    }
+
+    function repay(address usr, uint currencyAmount) public note auth {
+        reserve.repay(usr, currencyAmount);
+        quant.updateDebt(int(currencyAmount) * -1);
+        updateSupplyRate();
+    }
+
+    function borrow(address usr, uint currencyAmount) public note auth {
+        reserve.borrow(usr, currencyAmount);
+        quant.updateDebt(int(currencyAmount));
+        updateSupplyRate();
+    }
+
+    function adjustQuant(uint speed) public note auth {
+        quant.file("iBorrow", speed);
+        updateSupplyRate();
+    }
+
+    function updateSupplyRate() internal {
+        slicer.updateISupply(quant.getSpeed(), quant.debt(), reserve.balance());
```

Comment:
Let's rename `speed` into `rate`
```suggestion
        slicer.updateISupply(quant.borrowRate(), quant.debt(), reserve.balance());
```

Reply:
Should we not add here some flag if we should update the slicer based on `borrowRate` or the quant based on `supplyRate`?